sdk/rust/oak_config_serializer/src/main.rs

```diff
@@ -78,13 +87,46 @@ impl Default for InitialNodeConfig {
     }
 }
 
+fn get_sha256(data: &[u8]) -> Vec<u8> {
+    let mut hasher = Sha256::new();
+    hasher.update(data);
+    hasher.finalize().as_slice().to_vec()
+}
+
 /// Load Wasm module from URL or a file, if URL is not specified.
-fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
+async fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
     match module {
-        Module::Url(_url) => unimplemented!(),
         Module::Path(path) => {
             fs::read(&path).with_context(|| format!("Couldn't read file {}", path))
         }
+        Module::External(external) => {
+            let expected_sha256 =
+                hex::decode(&external.sha256).context("Couldn't decode SHA256 sum")?;
+            let url: Url = format!("{}/{}", &external.url, &external.sha256)
```

Comment:
Wait, so is it not using the actual provided URL? I think it should use whatever it's in the manifest, and then use the SHA as a checksum. Otherwise it becomes impossible to reuse modules that have been published with different names, right?

Reply:
Currently the name of the file is it's hash.
And the directory is called `database_proxy` for example.

Reply:
I think the URL should be used as it is. If the uploader wants to use the hash of the module as file name, they can, but they should not be forced to do it.

Reply:
Done